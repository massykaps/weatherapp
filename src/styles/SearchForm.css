@media (max-width: 768px) {
    .search-form {
      display: flex;
      justify-content: center;
    }
  }