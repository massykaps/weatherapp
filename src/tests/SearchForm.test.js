import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import SearchForm from "../components/SearchForm";

describe("SearchForm", () => {
  const validProps = {
    searchText: "Test Search Text",
    setSearchText: jest.fn(),
    onSubmit: jest.fn(),
  };

  test("renders", () => {
    const { asFragment } = render(
      <SearchForm
        searchText={validProps.searchText}
        setSearchText={validProps.setSearchText}
        onSubmit={validProps.onSubmit}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test("button displays correct text", () => {
    render(
      <SearchForm
        searchText={validProps.searchText}
        setSearchText={validProps.setSearchText}
        onSubmit={validProps.onSubmit}
      />
    );

    const button = screen.getByText("Search");

    expect(button).toBeInTheDocument();
  });

  test("clicking button causes submit function to be called", () => {
    render(
      <SearchForm
        searchText={validProps.searchText}
        setSearchText={validProps.setSearchText}
        onSubmit={validProps.onSubmit}
      />
    );
    const button = screen.getByText("Search");
    fireEvent.click(button);

    expect(validProps.onSubmit).toHaveBeenCalledTimes(1);
  });
});
