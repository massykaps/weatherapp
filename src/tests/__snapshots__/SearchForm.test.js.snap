// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm renders 1`] = `
<DocumentFragment>
  <div
    class="search-form"
  >
    <input
      type="text"
      value="Test Search Text"
    />
    <button
      type="submit"
    >
      Search
    </button>
  </div>
</DocumentFragment>
`;
